packages/react-button/src/components/Button/ButtonBase.tsx

```diff
@@ -31,8 +34,10 @@ export const ButtonBase = compose<'button', ButtonProps, ButtonProps, {}, {}>(
       'loading',
       'primary',
       'secondary',
+      'tokens',
       'size',
-    ],
+      // tslint:disable-next-line:no-any
+    ] as any,
```

Comment:
This needs to be discussed. `handledProps` is a mess with hooks.

I needed to add `buttonRef` to the state. `handledProps` only takes in `TProps`. The `compose` typings don't even take in `TState`, but even if it did, the hook is what knows about the state being returned and should own the state. So I'm perplexed here on what the right thing should be.

I am very much thinking that getNativeProps might need to be used, and perhaps explicitly called in each component.

Reply:
OK I've merged a getNativeProps performance improvement, so I'll revisit this in a separate PR.